modules/solid_mechanics/tests/j_integral_thermal/interaction_integral_2d.i

```diff
@@ -0,0 +1,230 @@
+#This problem from [Wilson 1979] tests the thermal strain term in the
```

Comment:
We should probably rename the `j_integral_thermal` directory to `domain_integral_thermal`.